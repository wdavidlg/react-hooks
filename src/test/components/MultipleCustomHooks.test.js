const { shallow } = require("enzyme")
const { MultipleCustomHooks } = require("../../components/03-examples/MultipleCustomHooks");
const { useCounter } = require("../../hooks/useCounter");
const { useFetch } = require("../../hooks/useFetch");
jest.mock("../../hooks/useFetch");
jest.mock('../../hooks/useCounter');

describe('Pruebas en <MultipleCustomHooks />', () => {

    beforeEach(() => {
        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {}
        });
    })
    
    test('debe mostrarse correctamente', () => {
        const state = {
            data: null,
            loading: true,
            error: null
        }
        useFetch.mockReturnValue(state);
        const wrapper = shallow(<MultipleCustomHooks/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar la informacion', () => {
        const data = [
            {
                quote_id: '123',
                quote: 'Hola mundo',
                author: 'Mark'
            }
        ]
        
        const state = {
            data,
            loading: false,
            error: null
        }
        useFetch.mockReturnValue(state);
        const wrapper = shallow(<MultipleCustomHooks/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('p').at(2).text()).toBe('Mark');
        
    })
    
    

})
