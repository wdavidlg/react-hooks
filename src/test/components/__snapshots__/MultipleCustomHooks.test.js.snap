// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> debe de mostrar la informacion 1`] = `
<div>
  <h1>
    The Breaking Bad Quotes
  </h1>
  <hr />
  <blockquote
    className="blockquote text-end"
  >
    <p>
      123
    </p>
    <p>
      Hola mundo
    </p>
    <p
      className="blockquote-footer"
    >
      Mark
    </p>
  </blockquote>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next Quote
  </button>
</div>
`;

exports[`Pruebas en <MultipleCustomHooks /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    The Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-success text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next Quote
  </button>
</div>
`;
